chia/util/full_block_utils.py

```diff
@@ -0,0 +1,208 @@
+from typing import Optional, Callable
+
+from chia.types.blockchain_format.program import SerializedProgram
+from clvm_rs import serialized_length
+from blspy import G1Element, G2Element
+
+
+def skip_list(buf: memoryview, skip_item: Callable[[memoryview], memoryview]) -> memoryview:
+    n = int.from_bytes(buf[:4], "big", signed=False)
+    buf = buf[4:]
+    for i in range(n):
+        buf = skip_item(buf)
+    return buf
+
+
+def skip_bytes(buf: memoryview) -> memoryview:
+    n = int.from_bytes(buf[:4], "big", signed=False)
+    buf = buf[4:]
+    assert n >= 0
+    return buf[n:]
+
+
+def skip_optional(buf: memoryview, skip_item: Callable[[memoryview], memoryview]) -> memoryview:
+
+    if buf[0] == 0:
+        return buf[1:]
+    assert buf[0] == 1
+    return skip_item(buf[1:])
+
+
+def skip_bytes32(buf: memoryview) -> memoryview:
+    return buf[32:]
+
+
+def skip_uint32(buf: memoryview) -> memoryview:
+    return buf[4:]
+
+
+def skip_uint64(buf: memoryview) -> memoryview:
+    return buf[8:]
+
+
+def skip_uint128(buf: memoryview) -> memoryview:
+    return buf[16:]
+
+
+def skip_uint8(buf: memoryview) -> memoryview:
+    return buf[1:]
+
+
+def skip_bool(buf: memoryview) -> memoryview:
+    assert buf[0] in [0, 1]
+    return buf[1:]
+
+
+# def skip_class_group_element(buf: memoryview) -> memoryview:
```

Comment:
Why comment this out?

Reply:
I made a few small optimizations by not calling some trivial functions, but just skip a fixed number of bytes instead. As you can see by the `skip_vdf_info()` function, it doesn't make those calls that are commented out, it just skips the bytes instead.

python doesn't seem to inline functions, so this abstraction appear to come at a price, and it doesn't seem very valuable.